dpki/src/seed.rs

```diff
@@ -227,6 +245,18 @@ impl DeviceSeed {
         pw_hash(pin, &mut self.inner.buf, &mut hash, config)?;
         Ok(DevicePinSeed::new(hash))
     }
+
+    /// Generate Auth Seed
+    /// @param {number} index - the index number in this seed group, must not be zero
+    pub fn generate_auth_seed(
+        &mut self,
+        index: u64,
+    ) -> HcResult<AuthSeed> {
+        let seed_context = SeedContext::new(AUTH_CTX);
+        let seed_buf =
+            generate_derived_seed_buf(&mut self.inner.buf, &seed_context, index, SEED_SIZE)?;
+        Ok(AuthSeed::new(seed_buf))
```

Comment:
It is overkill but we could introduce an enum for the request and result in the end we have

fn generate_seed(&mut self, index:i64, SeedType) -> SeedResult; 

where SeedType contains Auth, Revocation variants and the SeedResult contains the same, however not necessary if they are only two imo but could be a possibility if they are more